datafusion/core/tests/dataframe/mod.rs

```diff
@@ -2646,3 +2646,58 @@ async fn boolean_dictionary_as_filter() {
 
     assert_batches_eq!(expected, &result_df.collect().await.unwrap());
 }
+
+#[tokio::test]
```

Comment:
Thanks @jonahgao, nice PR, I would also add a test with some edge cases like nested alias, empty string alias

Reply:
Added in https://github.com/apache/datafusion/pull/14127/commits/af5136b5b3ffc1bc8216c552d46433cfa467c1ca